aws/deregister_amis.py

```diff
@@ -0,0 +1,105 @@
+"""
+Use boto to deregister AMIs that match a given tag key-value pair.
+
+That tag key-value pair is hardcoded to:
+    delete_or_keep: delete
+
+Usage:
+    If you've defined the AWS credentials as environment variables or in a
+    .boto file, then use:
+        `python deregister_amis.py
+
+    Else, you can add the aws keys as arguments to the above command:
+        ** '--aws-access-key-id' or '-i'
+        ** '--aws-secret-access-key' or '-s'
+
+    If you don't want to deregister AMIs, but you'd like to know which ones
+    you'd deregister if you ran the command, then use the --dry-run switch.
+
+"""
+import argparse
+import boto
+import logging
+import os
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+def get_ec2_connection(aws_access_key_id, aws_secret_access_key):
+    return boto.connect_ec2(aws_access_key_id, aws_secret_access_key)
```

Comment:
Wrap this in a try/catch block to help the user troubleshoot bad credential errors.